core-bundle/src/DependencyInjection/Configuration.php

```diff
@@ -514,10 +515,16 @@ private function getLocales(): array
         $languages = [$this->defaultLocale];
 
         /** @var array<SplFileInfo> $finder */
-        $finder = Finder::create()->directories()->depth(0)->name('/^[a-z]{2}(_[A-Z]{2})?$/')->in($dirs);
+        $finder = Finder::create()->directories()->depth(0)->name('/^[a-z]{2,}/')->in($dirs);
 
         foreach ($finder as $file) {
-            $languages[] = $file->getFilename();
+            $locale = $file->getFilename();
+
+            if (LocaleUtil::canonicalize($locale) !== $locale) {
+                continue;
```

Comment:
This means we are ignoring language folders such as `en_US.utf8` or `en_US@utf8`, right?

Reply:
Correct. We could never use them if they are named `en_US.utf8` because you cannot enter this in the backend anyways.

Reply:
That is why we previously only allowed the `xx` and `xx_XX` formats.